File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/checkout/CheckOutViewHolder.java
package com.example.myapplication.ui.fragment.checkout;

import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.myapplication.databinding.CardviewCheckoutBinding;
import com.example.myapplication.model.Product;
import com.example.myapplication.util.android.base.BaseViewHolder;

public class CheckOutViewHolder extends BaseViewHolder<CardviewCheckoutBinding, Product> {
    CheckOutViewHolder(@NonNull CardviewCheckoutBinding binding) {
        super(binding);
    }

    @Override
    public void bindTo(@Nullable Product item) {
        binding.getRoot().setOnClickListener(v ->
                Toast.makeText(v.getContext(), item != null ? item.toString() : "View clicked", Toast.LENGTH_SHORT).show());
    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/checkout/CheckOutViewHolder.kt
package com.example.myapplication.ui.fragment.checkout

import com.example.myapplication.R
import com.example.myapplication.databinding.CardviewCheckoutBinding
import com.example.myapplication.model.Product
import com.example.myapplication.util.android.base.BaseViewHolder

internal class CheckOutViewHolder(binding: CardviewCheckoutBinding) : BaseViewHolder<CardviewCheckoutBinding, Product>(binding) {
    override fun bindTo(item: Product) {
        binding.checkoutName.text = item.name
        binding.checkoutPrice.text = binding.root.context.getString(R.string.set_price, item.price)
        binding.checkoutQuantity.text = item.quantity.toString()
    }
}